Signal/src/view controllers/AttachmentSharing.m

```diff
@@ -10,29 +10,18 @@ @implementation AttachmentSharing
 + (void)showShareUIForAttachment:(TSAttachmentStream *)stream {
     OWSAssert(stream);
 
-    NSString *filePath = stream.filePath;
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSError *error;
-        NSData *data = [NSData dataWithContentsOfFile:filePath options:0 error:&error];
-        if (!data || error) {
-            DDLogError(@"%@ %s could not read data from attachment: %@.",
-                       self.tag,
-                       __PRETTY_FUNCTION__,
-                       error);
-            return;
-        }
-
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [AttachmentSharing showShareUIForData:data];
-        });
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [AttachmentSharing showShareUIForURL:stream.mediaURL];
```

Comment:
The file extension is what UIActivityViewController uses to determine which kinds of sharing are appropriate, so we should pass it a URL (which has a file extension) rather than just an NSData.